Sources/AlgoliaSearch-Client/Index.swift

```diff
@@ -407,12 +407,20 @@ import Foundation
     ///
     @discardableResult public func searchForFacetValues(of facetName: String, matching text: String, query: Query? = nil, requestOptions: RequestOptions? = nil, completionHandler: @escaping CompletionHandler) -> Operation {
         let path = "1/indexes/\(urlEncodedName)/facets/\(facetName.urlEncodedPathComponent())/query"
-        let params = query != nil ? Query(copy: query!) : Query()
+      if let params = query {
         params["facetQuery"] = text
         let requestBody = [
-            "params": params.build()
+          "params": params.build()
         ]
         return client.performHTTPQuery(path: path, method: .POST, body: requestBody, hostnames: client.readHosts, isSearchQuery: true, requestOptions: requestOptions, completionHandler: completionHandler)
+      } else {
+        let params = Query()
+        params["facetQuery"] = text
+        let requestBody = [
+          "params": params.build()
+        ]
+        return client.performHTTPQuery(path: path, method: .POST, body: requestBody, hostnames: client.readHosts, isSearchQuery: true, requestOptions: requestOptions, completionHandler: completionHandler)
```

Comment:
Why not taking this out of if / else since is the same call

Reply:
Yes I can. I'll have to move the `requestBody` out as well, I'll do this